js/controllers/todos_controller.js

```diff
@@ -1,4 +1,125 @@
+                } else {
+                    me.get('model').save();
+                }
+            }), 100);
+        },
+        removeTask: function () {
+            var Task = this.get('model');
+            Task.deleteRecord();
+            Task.save();
+        }
+    },
+
+    editMode: false,
+
+    isDone: function (key, value) {
+        var model = this.get('model');
+
+        if (value === undefined) {
+            // property being used as a getter
+            return model.get('isDone');
+        } else {
+            // property being used as  setter
+            model.set('isDone', value);
+            model.save();
+            return value;
+        }
+    }.property('model.isDone')
+});
+
+Abs.AbsController = Ember.ArrayController.extend({
+    actions: {
+        checkAll: function () {
+            var checked = $('#chk-all')[0].checked;
+            var store = this.get('store'), model = this.get('model');
+
+            model.forEach(function (task) {
+                task.set('isDone', checked);
+                task.save();
+            });
+
+        },
+        createTask: function () {
+            // Get the Tasks title set by the "New Tasks" text field
+            var title = this.get('newTitle');
+            if (!title.trim()) { return; }
+
+            // Create the new Tasks model
+            var Tasks = this.store.createRecord('Tasks', {
+                title: title,
+                isDone: false
+            });
+
+            // Clear the "New Tasks" text field
+            this.set('newTitle', '');
+
+            // Save the new model
+            Tasks.save();
+        },
+
+        removeAllCompleted: function () {
+            var model = this.get('model');
+            model.forEach(function (task) {
+                if (task && task.get('isDone')) {
+                    Ember.run.once(this, function () {
+                        task.deleteRecord();
+                        task.save();
+                    });
+                }
+
+            });
+
+        },
+
+        allTasks: function () {
+            var me = this, store = me.get('store');
+            me.set('model', store.find('Tasks'));
+            this.removeSelectedClass();
+            $('#allTasksBtn').addClass('selected');
+        },
+        doneTasks: function () {
+            var me = this, store = me.get('store');
+            me.set('model', store.filter('Tasks', function (Tasks) {
+                return Tasks.get('isDone');
+            }));
+            this.removeSelectedClass();
+            $('#doneTasksBtn').addClass('selected');
+        },
+        activeTasks: function () {
+            var me = this, store = me.get('store');
+            me.set('model', store.filter('Tasks', function (Tasks) {
+                return Tasks.get('isDone') == false;
+            }));
+            this.removeSelectedClass();
+            $('#activeTasksBtn').addClass('selected');
+        },
```

Comment:
these functions are for show completed and not completed tasks, they can combine and write to a new method by a parameter
        showTasks: function (isActive) {
> +            var me = this, store = me.get('store');
> +            me.set('model', store.filter('Tasks', function (Tasks) {
> +                return Tasks.get('isDone') == isActive;
> +            }));
> +            this.removeSelectedClass();
> +            $('#activeTasksBtn').addClass('selected');
> +        }, 


     On Monday, September 28, 2015 6:00 PM, Ruan Naude <notifications@github.com> wrote:
   

 In js/controllers/todos_controller.js:> +        doneTasks: function () {
> +            var me = this, store = me.get('store');
> +            me.set('model', store.filter('Tasks', function (Tasks) {
> +                return Tasks.get('isDone');
> +            }));
> +            this.removeSelectedClass();
> +            $('#doneTasksBtn').addClass('selected');
> +        },
> +        activeTasks: function () {
> +            var me = this, store = me.get('store');
> +            me.set('model', store.filter('Tasks', function (Tasks) {
> +                return Tasks.get('isDone') == false;
> +            }));
> +            this.removeSelectedClass();
> +            $('#activeTasksBtn').addClass('selected');
> +        },
,?—
Reply to this email directly or view it on GitHub.